sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -405,7 +405,17 @@ class MapboxMap :
    */
   override fun setCamera(cameraOptions: CameraOptions) {
     checkNativeMap("setCamera")
-    nativeMap.setCamera(cameraOptions)
+    try {
+      nativeMap.setCamera(cameraOptions)
+    } catch (thr: Throwable) {
+      logW(
+        TAG,
+        """
+          setCamera failed: ${thr.message}
+          This will happen when using com.mapbox.maps.BackgroundAnimatorThread 
+        """.trimIndent()
+      )
+    }
```

Comment:
we should not call setCamera from background thread even in AA case for Xiaomi. All animations should go thru animation plugin where this is already handled

Reply:
Agree with a larger refactor or improvement. This try catch is the simplest possible solution, in order to try and make it work.

Reply:
Looking to ensure that `indicatorPositionChangedListener` will only be posted by the main thread, then this try/catch could be removed.

That is this crash https://github.com/mapbox/mapbox-maps-android/pull/1605#issuecomment-1217969582

Reply:
This will crash even when it's posted from the main thread. There are more fixes needed in order to remove this try/catch. Note that this crash happens even when the ThreadChecker is false

```
2022-08-18 11:21:18.203 5923-5923/com.mapbox.maps.testapp.auto I/Mapbox: [maps-android\kyle_debug]: OnIndicatorPositionChangedListener main
2022-08-18 11:21:18.219 5923-5923/com.mapbox.maps.testapp.auto I/Mapbox: [maps-android\kyle_debug]: OnIndicatorPositionChangedListener main
2022-08-18 11:21:18.236 5923-5923/com.mapbox.maps.testapp.auto I/Mapbox: [maps-android\kyle_debug]: OnIndicatorPositionChangedListener main
2022-08-18 11:21:18.353 5923-6192/com.mapbox.maps.testapp.auto E/AndroidRuntime: FATAL EXCEPTION: animation_thread_aee6cc7a-6e7c-41a6-8790-0337587b8d24
    Process: com.mapbox.maps.testapp.auto, PID: 5923
    java.lang.Error: std::exception
        at com.mapbox.maps.CameraManager.setCamera(Native Method)
        at com.mapbox.maps.NativeMapImpl.setCamera(NativeMapImpl.kt:36)
        at com.mapbox.maps.MapboxMap.setCamera(MapboxMap.kt:408)
        at com.mapbox.maps.testapp.auto.car.CarCameraController.changeBearingListener$lambda-3(CarCameraController.kt:47)
        at com.mapbox.maps.testapp.auto.car.CarCameraController.$r8$lambda$FSsysefwAVT1AjEVdrjGVZ0YE9k(Unknown Source:0)
        at com.mapbox.maps.testapp.auto.car.CarCameraController$$ExternalSyntheticLambda0.onIndicatorBearingChanged(Unknown Source:2)
        at com.mapbox.maps.plugin.locationcomponent.LocationComponentPluginImpl.indicatorBearingChangedListener$lambda-5(LocationComponentPluginImpl.kt:146)
        at com.mapbox.maps.plugin.locationcomponent.LocationComponentPluginImpl.$r8$lambda$ENafjuhdBpO7PRaFQ5TOHHc_Y54(Unknown Source:0)
        at com.mapbox.maps.plugin.locationcomponent.LocationComponentPluginImpl$$ExternalSyntheticLambda2.onIndicatorBearingChanged(Unknown Source:2)
        at com.mapbox.maps.plugin.locationcomponent.animators.PuckBearingAnimator.updateLayer(PuckBearingAnimator.kt:12)
        at com.mapbox.maps.plugin.locationcomponent.animators.PuckBearingAnimator.updateLayer(PuckBearingAnimator.kt:5)
        at com.mapbox.maps.plugin.locationcomponent.animators.PuckAnimator._init_$lambda-0(PuckAnimator.kt:34)
        at com.mapbox.maps.plugin.locationcomponent.animators.PuckAnimator.$r8$lambda$DKuRutUBCyf6yKd5UEMf-U8meWw(Unknown Source:0)
        at com.mapbox.maps.plugin.locationcomponent.animators.PuckAnimator$$ExternalSyntheticLambda0.onAnimationUpdate(Unknown Source:2)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1566)
        at android.animation.ValueAnimator.animateBasedOnTime(ValueAnimator.java:1357)
        at android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1489)
        at android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)
        at android.animation.AnimationHandler.access$100(AnimationHandler.java:37)
        at android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1140)
        at android.view.Choreographer.doCallbacks(Choreographer.java:946)
        at android.view.Choreographer.doFrame(Choreographer.java:870)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1127)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:210)
        at android.os.Looper.loop(Looper.java:299)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```

Reply:
```
    Process: com.mapbox.maps.testapp.auto, PID: 5923
    java.lang.Error: std::exception
        at com.mapbox.maps.CameraManager.setCamera(Native Method)
```
gl-native team is going to catch this exception on their end in the next few releases, so we could skip this crash and catch it for now.